### PR TITLE
build: Switch to using c++17

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**

### CODE REVIEW

**Comment by jeremyg-lunarg** [comment]
@juan-lunarg Was there anything else done in VVL to support the switch that I should pull over?

File: BUILD.md [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 1. Python >= 3.7 (3.6 may work, 3.5 and earlier is not supported)
 1. CMake >= 3.10.2
-1. C++ >= c++11 compiler. See platform-specific sections below for supported compiler versions.
+1. C++ >= c++17 compiler. See platform-specific sections below for supported compiler versions.
```
Comment (juan-lunarg) [inline BUILD.md:19]: Line 482 also has a mention of C++11

"This repository has been built and tested on the two most recent Ubuntu LTS
versions. Currently, the oldest supported version is Ubuntu 16.04, meaning
that the minimum officially supported C++11 compiler version is GCC 5.4.0,
although earlier versions may work. It should be straightforward to adapt this
repository to other Linux distributions."

**Review by juan-lunarg — CHANGES_REQUESTED** [review]

**Comment by juan-lunarg** [comment]
> @juan-lunarg Was there anything else done in VVL to support the switch that I should pull over?

Besides my minor nitpick this looks good. The main trouble was getting the chromium build to work.

**Review by juan-lunarg — APPROVED** [review]

**Review by juan-lunarg — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -108,8 +108,14 @@ set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 # "Helper" targets that don't have interesting source code should set their FOLDER property to this
 set(LAYERS_HELPER_FOLDER "Helper Targets")
 
+# NOTE: The idiom for open source projects is to not enable warnings as errors.
+# This reduces problems for users who simply want to build the repository.
+option(BUILD_WERROR "Treat compiler warnings as errors" OFF)
```
[the remark this answers is not in the document]
Comment (jeremyg-lunarg) [reply CMakeLists.txt:113]: thanks. you already brought over the macos part, I updated common_ci.py to match

**Review by juan-lunarg — APPROVED** [review]

**Pull request merged**